astropy/visualization/interval.py

```diff
@@ -6,39 +6,73 @@
 """
 
 from __future__ import division, print_function
-
 import abc
 import numpy as np
-
+from ..extern import six
+from ..utils.misc import InheritDocstrings
 from .transform import BaseTransform
 from .zscale import zscale
 
+
 __all__ = ['BaseInterval', 'ManualInterval', 'MinMaxInterval',
-           'PercentileInterval', 'AsymmetricPercentileInterval',
+           'AsymmetricPercentileInterval', 'PercentileInterval',
            'ZScaleInterval']
 
 
+@six.add_metaclass(InheritDocstrings)
 class BaseInterval(BaseTransform):
     """
-    Base class for the interval classes, which, when called with an array of
-    values, return an interval computed following different algorithms.
+    Base class for the interval classes, which, when called with an
+    array of values, return an interval computed following different
+    algorithms.
     """
 
     @abc.abstractmethod
     def get_limits(self, values):
         """
-        Return the minimum and maximum value in the interval based on the values provided.
+        Return the minimum and maximum value in the interval based on
+        the values provided.
+
+        Parameters
+        ----------
+        values : `~numpy.ndarray`
+            The image values.
+
+        Returns
+        -------
+        vmin, vmax : float
+            The mininium and maximum image value in the interval.
         """
 
     def __call__(self, values, clip=True, out=None):
+        """
+        Transform values using this interval.
+
+        Parameters
+        ----------
+        values : array-like
+            The input values.
+        clip : bool, optional
+            If `True` (default), values outside the [0:1] range are
+            clipped to the [0:1] range.
+        out : `~numpy.ndarray`, optional
+            If specified, the output values will be placed in this array
+            (typically used for in-place calculations).
+
+        Returns
+        -------
+        result : `~numpy.ndarray`
+            The transformed values.
+        """
 
         vmin, vmax = self.get_limits(values)
 
         if out is None:
             values = np.subtract(values, float(vmin))
         else:
             if out.dtype.kind != 'f':
-                raise TypeError("Can only do in-place scaling for floating-point arrays")
+                raise TypeError('Can only do in-place scaling for '
```

Comment:
Somwhere it was mentioned that error messages should start with a lowercase. Don't remember where though.